x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/job/DateHistogramGroupConfig.java

```diff
@@ -47,71 +52,80 @@
+    public static final String TIME_ZONE = "time_zone";
+    private static final String DELAY = "delay";
+    private static final String DEFAULT_TIMEZONE = "UTC";
+    private static final ConstructingObjectParser<DateHistogramGroupConfig, Void> PARSER;
+    static {
+        PARSER = new ConstructingObjectParser<>(NAME, a ->
+            new DateHistogramGroupConfig((String) a[0], (DateHistogramInterval) a[1], (DateHistogramInterval) a[2], (String) a[3]));
+        PARSER.declareString(constructorArg(), new ParseField(FIELD));
+        PARSER.declareField(constructorArg(), p -> new DateHistogramInterval(p.text()), new ParseField(INTERVAL), ValueType.STRING);
+        PARSER.declareField(optionalConstructorArg(),  p -> new DateHistogramInterval(p.text()), new ParseField(DELAY), ValueType.STRING);
+        PARSER.declareString(constructorArg(), new ParseField(TIME_ZONE));
+    }
 
-    private final DateHistogramInterval interval;
     private final String field;
-    private final DateTimeZone timeZone;
+    private final DateHistogramInterval interval;
     private final DateHistogramInterval delay;
+    private final String timeZone;
+
+    public DateHistogramGroupConfig(final String field,
+                                    final DateHistogramInterval interval,
+                                    final @Nullable DateHistogramInterval delay,
+                                    final String timeZone) {
+        if (field == null || field.isEmpty()) {
+            throw new IllegalArgumentException("Field must be a non-null, non-empty string");
+        }
+        if (interval == null) {
+            throw new IllegalArgumentException("Interval must be a non-null, non-empty string");
+        }
 
-    static {
-        PARSER.declareField(DateHistoGroupConfig.Builder::setInterval,
-                p -> new DateHistogramInterval(p.text()), INTERVAL, ObjectParser.ValueType.STRING);
-        PARSER.declareString(DateHistoGroupConfig.Builder::setField, FIELD);
-        PARSER.declareField(DateHistoGroupConfig.Builder::setDelay,
-                p -> new DateHistogramInterval(p.text()), DELAY, ObjectParser.ValueType.LONG);
-        PARSER.declareField(DateHistoGroupConfig.Builder::setTimeZone, p -> {
-            if (p.currentToken() == XContentParser.Token.VALUE_STRING) {
-                return DateTimeZone.forID(p.text());
-            } else {
-                return DateTimeZone.forOffsetHours(p.intValue());
-            }
-        }, TIME_ZONE, ObjectParser.ValueType.LONG);
-    }
-
-    private DateHistoGroupConfig(DateHistogramInterval interval,
-                                 String field,
-                                 DateHistogramInterval delay,
-                                 DateTimeZone timeZone) {
         this.interval = interval;
         this.field = field;
         this.delay = delay;
-        this.timeZone = Objects.requireNonNull(timeZone);
+        this.timeZone = timeZone != null ? timeZone : DEFAULT_TIMEZONE;
```

Comment:
Since timezone is now a string, we should probably check for `Strings.isNullOrEmpty()` instead of just null now.